snapshot-labs/sx-monorepo
# feat: display oSnap execution info

### Summary

This PR adds message about oSnap execution progress.

Because oSnap executions are per-treasury (and not per-proposal) design had to be updated to move execution status and actions into per-treasury sections.

Closes: https://github.com/snapshot-labs/sx-monorepo/issues/467

### How to test

1. Check proposals at different stages:
2. Waiting to be confirmed: http://localhost:8080/#/s:0cf5e.eth/proposal/0x0a04e2d340fcd5cfcac5d89369d152c8c9fefc59126c48b650785c147dc3d6aa
3. Executed: http://localhost:8080/#/s:0cf5e.eth/proposal/0x2a0752ef99d948dc753fb4b5359fe34a352aaf3c9f10b8ee5edf9ad4cde9246c
4. (CHECK COMMENT BELOW) Non compliant: http://localhost:8080/#/s:fabien.eth/proposal/0xeebb4a2cb25a84de837135b9be63246b1f78c1b351edc63a00652de0e419a4f3

Currently `osnap.uma.xyz` rejects requests from outside snapshot.org due to its CORS configuration. That means that checking for compliance will fail at the moment until it's addressed on their side (and message of `Execution details failed to load.` will be displayed). You can mock it during testing using this patch:
```diff
diff --git a/apps/ui/src/composables/useExecutionActions.ts b/apps/ui/src/composables/useExecutionActions.ts
index 486644a0..f7fd2225 100644
--- a/apps/ui/src/composables/useExecutionActions.ts
+++ b/apps/ui/src/composables/useExecutionActions.ts
@@ -142,14 +142,15 @@ export function useExecutionActions(
       });
 
       if (!data) {
-        const configCompliant = await isConfigCompliant(
-          execution.safeAddress,
-          execution.chainId
-        );
-
-        message.value = configCompliant.automaticExecution
-          ? 'Waiting for execution to be initiated.'
-          : 'Space is not configured for automatic execution.';
+        message.value = 'Space is not configured for automatic execution.';
+        // const configCompliant = await isConfigCompliant(
+        //   execution.safeAddress,
+        //   execution.chainId
+        // );
+
+        // message.value = configCompliant.automaticExecution
+        //   ? 'Waiting for execution to be initiated.'
+        //   : 'Space is not configured for automatic execution.';
       } else if (data.executionTransactionHash) {
         try {
           executionNetwork.value = getNetwork(execution.networkId);

```

### Screenshots

Space is misconfigured for automatic execution:
<img width="598" alt="image" src="https://github.com/user-attachments/assets/1cbee158-905a-485a-a6ea-c68048e4314a">

Proposal automatic execution was initiated:
<img width="597" alt="image" src="https://github.com/user-attachments/assets/53749360-2ed6-4578-9bdf-bae5579fa43c">


Proposal has been executed:
<img width="616" alt="image" src="https://github.com/user-attachments/assets/0de41be1-7b8d-408b-96cf-78bd841eddf2">
